### PR TITLE
Added support for --dart-define

### PR DESCRIPTION
**Pull request by alenros**
Added support for --dart-define

### CODE REVIEW

**Comment by hey24sheep** [comment]
@alenros can you resolve conflicts and push. Also, could you add this to readme as well.

**Comment by chadrickman** [comment]
Will this only support a single variable not multiples being passed in? EX: `--dart-define=SOME_VAR=SOME_VALUE --dart-define=OTHER_VAR=OTHER_VALUE`

**Comment by alenros** [comment]
@chadrickman you are right that it supports just one.

@hey24sheep How about separating multiple values by a comma?

**Comment by hey24sheep** [comment]
@alenros I agree, it would be best to use comma separation.

File: tasks/build/index.ts [hunk 1]
```diff
@@ -28,18 +28,19 @@ async function main(): Promise<void> {
     let buildNumber = task.getInput('buildNumber', false);
     let buildFlavour = task.getInput('buildFlavour', false);
     let entryPoint = task.getInput('entryPoint', false);
+    let dartDefine = task.getInput('dartDefine', false);
```
Comment (hey24sheep) [inline tasks/build/index.ts:31]: You could use 'getDelimitedInput' instead to get string[] 'comma separated vals'

File: tasks/build/index.ts [hunk 2]
```diff
@@ -91,6 +93,10 @@ async function buildApk(
     if (entryPoint) {
         args.push("--target=" + entryPoint);
     }
+    
+     if (dartDefine) {
+        args.push("--dart-define="+ dartDefine);
```
Comment (hey24sheep) [inline tasks/build/index.ts:98]: loop and add args. Make sure comma separation doesn't include strings like 'some_var=some_value, --dart-define='"

if --dart-define is given, remove it before pushing

File: tasks/build/task.json [hunk 3]
```diff
@@ -131,11 +131,19 @@
...
             "visibleRule": "target = ios"
-        }
+        },
+        {
+            "name": "dartDefine",
+            "type": "string",
+            "label": "Parameter value for --dart-define",
+            "defaultValue": "",
+            "required": false,
+            "helpMarkDown": "Parameter passed to --dart-define"
```
Comment (hey24sheep) [inline tasks/build/task.json:141]: "helpMarkDown" add ex: 'some_var=some_value, second_var=value, four=five"

so people don't include --dart-define again like 'some_var=some_value, --dart-define='". Though we will handle it as an edge case.

**Comment by hey24sheep** [comment]
@alenros @chadrickman Actually, I do not think we need to split it. args.push is string[] so end exec statement will be --dart-define=something=something --dart-define=s1=s2. So it should work. What do you think?

**Comment by hey24sheep** [comment]
@alenros I have merged this, If it doesn't work for multi args we will use delimited. It should work though. Thanks for your PR.

**Comment by alenros** [comment]
I ran a real build. Works nicely!

**Comment by hey24sheep** [comment]
@alenros nice, knew it should work. Thanks 👍

**Pull request merged**